Build: Make bwc builds always use snapshot=false

rjernst opened the pull request:
This commit makes the bwc builds use Exec instead of GradleBuild when
doing a release build. This is only relevant in the case of running
tests with a release build, which is only done by CI, so the sub build
in that case is run with info logging.

closes #26702

nik9000 commented:
Lgtm

On Mon, Sep 18, 2017, 11:36 PM Ryan Ernst <notifications@github.com> wrote:

> This commit makes the bwc builds use Exec instead of GradleBuild when
> doing a release build. This is only relevant in the case of running
> tests with a release build, which is only done by CI, so the sub build
> in that case is run with info logging.
>
> closes #26702 <https://github.com/elastic/elasticsearch/issues/26702>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/elastic/elasticsearch/pull/26704
> Commit Summary
>
>    - Build: Make bwc builds always use snapshot=false
>
> File Changes
>
>    - *M*
>    buildSrc/src/main/groovy/org/elasticsearch/gradle/VersionProperties.groovy
>    <https://github.com/elastic/elasticsearch/pull/26704/files#diff-0> (2)
>    - *M* distribution/bwc/build.gradle
>    <https://github.com/elastic/elasticsearch/pull/26704/files#diff-1>
>    (20)
>
> Patch Links:
>
>    - https://github.com/elastic/elasticsearch/pull/26704.patch
>    - https://github.com/elastic/elasticsearch/pull/26704.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/elastic/elasticsearch/pull/26704>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AANLomJafIDw8-k0rlmV5aDP_U_DlPtiks5sjzbagaJpZM4Pb0DQ>
> .
>

rjernst committed:
Build: Set bwc builds to always set snapshot

This commit enforces bwc builds always generate snapshot versions, even
when testing release versions in CI.

closes #26702

rjernst commented:
I updated this to override the sysprops via the startParameter of GradleBuild. I'm willing to use this for now, but I am apprehensive about it because I know it means gradle is using `System.setProperty` internally.
